articles/azure-monitor/app/tutorial-asp-net-core.md

```diff
@@ -0,0 +1,487 @@
+
+    ```kql
+    traces 
+    | where customDimensions.CategoryName == "ValuesController"
+    ```
+
+## Control the level of logs sent to Application Insights
+
+`ILogger` implementations have a built-in mechanism to apply [log filtering](https://docs.microsoft.com/dotnet/core/extensions/logging#how-filtering-rules-are-applied). This filtering lets you control the logs that are sent to each registered provider, including the Application Insights provider. You can use the filtering either in configuration (using an *appsettings.json* file) or in code. For more information about log levels and guidance on appropriate use, see the [Log Level](https://docs.microsoft.com/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0#log-level) documentation.
+
+The following examples show how to apply filter rules to the `ApplicationInsightsLoggerProvider` to control the level of logs sent to Application Insights.
+
+### Create filter rules with configuration
+
+The `ApplicationInsightsLoggerProvider` is aliased as **ApplicationInsights** in configuration. The following section of an *appsettings.json* file sets the default log level for all providers to <xref:Microsoft.Extensions.Logging.LogLevel.Warning?displayProperty=nameWithType>. The configuration for the ApplicationInsights provider specifically for categories that start with "ValuesController" override this default value with <xref:Microsoft.Extensions.Logging.LogLevel.Error?displayProperty=nameWithType> and higher.
+
+```json
+{
+  //... additional code removed for brevity
+  "Logging": {
+    "LogLevel": { // No provider, LogLevel applies to all the enabled providers.
+      "Default": "Warning"
+    },
+    "ApplicationInsights": { // Specific to the provider, LogLevel applies to the Application Insights provider.
+      "LogLevel": {
+        "ValuesController": "Error" //Log Level for the "ValuesController" category
+      }
+    }
+  }
+}
+```
+
+Deploying the sample application with the preceding code in *appsettings.json* will yield only the error trace being sent to Application Insights when interacting with the **ValuesController**. This is because the **LogLevel** for the **ValuesController** category is set to **Error**, therefore the **Warning** trace is suppressed.
+
+## Turn off logging to Application Insights
+
+To disable logging using configuration, set all LogLevel values to "None".
+
+```json
+{
+  //... additional code removed for brevity
+  "Logging": {
+    "LogLevel": { // No provider, LogLevel applies to all the enabled providers.
+      "Default": "None"
+    },
+    "ApplicationInsights": { // Specific to the provider, LogLevel applies to the Application Insights provider.
+      "LogLevel": {
+        "ValuesController": "None" //Log Level for the "ValuesController" category
```

Comment:
this is applicable to logs from "ValuesController" only.

Reply:
Do you want me to change anything in the code here? The text below this snippet explains the result is from the ValuesController only. Copy and paste: 

Deploying the sample application with the preceding code in *appsettings.json* will yield only the error trace being sent to Application Insights when interacting with the **ValuesController**. This is because the **LogLevel** for the **ValuesController** category is set to **Error**, therefore the **Warning** trace is suppressed.